tests/test_updater_with_simulator.py

```diff
@@ -257,6 +259,50 @@ def test_not_loading_targets_twice(self, wrapped_open: MagicMock) -> None:
         updater.get_targetinfo("somepath")
         wrapped_open.assert_not_called()
 
+    @patch.object(builtins, "open", wraps=builtins.open)
+    def test_expired_metadata(self, wrapped_open: MagicMock) -> None:
+        updater = self._run_refresh()
+
+        past_datetime = datetime.utcnow().replace(microsecond=0) - timedelta(days=5)
+
+        self.sim.timestamp.expires = past_datetime
+        self.sim.snapshot.expires = past_datetime
+        self.sim.targets.expires = past_datetime
+
+        # Add targets to repository
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b'some content', self.targets_dir)
+        self.sim.update_snapshot()
+
+        # Clean up calls to open during refresh()
+        wrapped_open.reset_mock()
+
+        # Make a successful update of valid metadata which stores it in cache
+        updater.get_targetinfo(self.targets_dir)
+
+        # Create a new updater and perform a second update while
+        # the metadata is already stored in cache (metadata dir)
+        updater = Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            self.sim,
+        )
+
+        with self.assertRaises(ExpiredMetadataError):
+            updater.get_targetinfo(self.targets_dir)
```

Comment:
Why do you call `updater.get_targetinfo(self.targets_dir)`?
Why don't you use directly `updater.refresh()`?
Also, the argument to `updater.get_targetinfo()` should be a path to a specific target, not the targets directory,

Reply:
I asked Teodora about this here https://github.com/theupdateframework/python-tuf/issues/1681#issuecomment-987330543,